### PR TITLE
Introduce Message Validator

### PR DESCRIPTION
**Pull request by cblgh**
so we had some chats about this in the dev cabal

i saw some patterns being repeated and figured we could extract the absolute useful minimum to a util function and save us some effort typing. when introducing it in the other views i felt that it also gave me a lot of extra headspace, since i didn't have to spend any effort thinking about the rest of the code working, so that might be added benefit of having a validator

but i felt we needed to have a pr to allow discussion as it didn't seem as if it was a 100% clear change (it actually sets a pretty hefty precedent for future messages, so let's get it right, even if it is a tiny change code-wise)

basically this change says that a message HAS TO have a `type`, it has to have a `content` key, and that content key has to point to an object so

```
{
  type: ...,
  content: {}
}

### CODE REVIEW

**Comment by nikolaiwarner** [comment]
I'm trying to think of all of the alternative messages one may want to send. Let's see if these all fit that model (some may be out of the scope of a modern team instant messaging application ❤️):

- chat/text
- a private message
- a thread reply
- a system message
- a image/video/audio file (with additional text message)
- any file (with additional text message)
- a dat / link to dat
- a reaction (emoji/thumbs up/etc)
- favorite/star a message
- a "shared" message (reference a message elsewhere)
- a voice/video call
- read receipts (mark that you've read a message)
- telemetry data (@noffle's gps location)
- vr object attributes (position, velocity, etc) :D

File: views/messages.js [hunk 1]
```diff
@@ -4,13 +4,14 @@ var readonly = require('read-only-stream')
...
 var EventEmitter = require('events').EventEmitter
+var isValidMessage = require("../util").isValidMessage
 
 module.exports = function (lvl) {
   var events = new EventEmitter()
 
   return View(lvl, {
     map: function (msg) {
-      if (!msg.value.timestamp) return []
+      if (!isValidMessage || !msg.value.timestamp) return []
```
Comment (noffle) [inline views/messages.js]: `s/isValidMessage/isValidMessage(msg)`

strongly recommend writing tests for this new logic!

**Comment by cblgh** [comment]
i'll revisit this, closing because it's stale
